packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -458,8 +458,12 @@ class JITSnapshotter {
       outputPaths.add(isolateSnapshotInstructions);
     }
 
+    // IA32 instructions generated by Dart VM can't be serialized yet,
+    // so for now let's create a snapshot with only the data section.
+    final bool supportsAppJit = (platform != TargetPlatform.android_x86);
```

Comment:
It's also the case that an x64 Windows host will produce an AppJIT snapshot that won't run on x64 Android (different ABI). x64 Mac/Linux/Fuchsia hosts are fine.

Reply:
Done